### PR TITLE
Add `RefreshMode` and `RenderMode` support to mobile navigator.

### PR DESCRIPTION
**Pull request by TomTirapani**
Addresses issue #1720

See Toolbox PR: https://github.com/xh/toolbox/pull/375

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: mobile/cmp/navigator/NavigatorPageModel.js [hunk 1]
```diff
@@ -34,26 +40,55 @@ export class NavigatorPageModel {
         return stringify({id, props}, qsOpts);
     }
 
+    get renderMode() {
+        return this._renderMode || this.navigatorModel.renderMode;
```
Comment (lbwexler) [inline mobile/cmp/navigator/NavigatorPageModel.js]: we should prefer nullish coalesce ('??') here (and I realize this code is copied from some similar objects)

**Comment by lbwexler** [comment]
This is great, and looks a lot like TabModel.  Would we consider pushing that analog even further in the naming?  Specifically, would we consider renaming NavigatorPageModel to just `PageModel` and have the `routes` argument to NavigatorModels ctor renamed  to `pages`;

**Comment by TomTirapani** [comment]
Thanks for taking a look. Great idea about the naming - will make those updates. My main concern there is that this makes it a breaking change, but it will be quite obviously broken and fairly trivial to update apps.

**Comment by lbwexler** [comment]
could potentially maintain "routes" as a deprecated name.   But I think
could also sell this as an enhancement worth retyping that one key.  Don't
think anything else likely to be breaking.

**Comment by TomTirapani** [comment]
Note that navigator already exposes an observable `pages[]`, which is distinct from routes. (routes being the "possible" pages and how they map to Router5, and pages being the instances of the pages currently in the stack)

**Comment by TomTirapani** [comment]
I've implemented the refactor as discussed yesterday with Lee. Please see new Toolbox PR required for this branch: https://github.com/xh/toolbox/pull/375

**Pull request merged**